pkg/extensions/sync/sync_test.go

```diff
@@ -227,6 +227,69 @@ func makeDownstreamServer(
 	return dctlr, destBaseURL, destDir, client
 }
 
+func makeDownstreamServerFixedPort(
```

Comment:
This function is a copy of `makeDownstreamServer` above and enhanced to include support for fixed ports (which is needed for scale out cluster) and also support configuring scale out cluster config.

There is some opportunity for code re-use here.

Reply:
If this is cluster related put this also in cluster pkg

Reply:
This is not exactly cluster specific. It just accepts a fixed port number (the existing one finds an available port) and accepts a cluster config to assign to the config struct.

There is an opportunity to refactor this, but I think I'd like to take it up later on.